lib/internal/fs/utils.js

```diff
@@ -331,6 +331,15 @@ function getOptions(options, defaultOptions) {
   if (options.signal !== undefined) {
     validateAbortSignal(options.signal, 'options.signal');
   }
+
+  if (defaultOptions.kPerRead) {
+    if (options.kPerRead === undefined) {
+      options.kPerRead = defaultOptions.kPerRead;
```

Comment:
Isn't this mutating the end user's `options` object?

Reply:
What I mean by this is that I believe `options` is the object passed in by the user. We shouldn't overwrite/set values on the object they pass in. AFAICT the only time `options` is not an end user's object is when they don't supply one.